src/Illuminate/Foundation/Testing/DatabaseTruncates.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Illuminate\Foundation\Testing;
+
+use Illuminate\Foundation\Testing\Traits\CanConfigureMigrationCommands;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+trait DatabaseTruncates
+{
+    use CanConfigureMigrationCommands;
+
+    protected static array $allTables;
+
+    protected function truncateTables()
+    {
+        // Always remove any test data before the application is destroyed.
+        $this->beforeApplicationDestroyed(function () {
+            Schema::disableForeignKeyConstraints();
+            collect(static::$allTables ??= DB::connection()->getDoctrineSchemaManager()->listTableNames())
+                ->diff($this->excludeTables())
+                ->filter(fn ($table) => DB::table($table)->exists())
+                ->each(fn ($table) => DB::table($table)->truncate());
+        });
```

Comment:
Don't you have to re-enable foreign keys after truncate?

Reply:
Only if the app uses them. I think this will need to use them by default but have the developer be able to configure it. I'll put something together.

Reply:
In all the testing I've done, the foreign keys constraints don't need to be manually re-enabled when the application is being destroyed. I'm happy to be proven wrong on this, as I have a configurable solution ready to go. However, for the sake of simplicity and speed I've left the `Schema::enableForeignKeyConstraints()` out.

To test: try to truncate a table with non-nullable foreign keys in a subsequent test.

```php
DB::table('users')->truncate();
```